tests/Unit/Analyzer/FileVisitorTest.php

```diff
@@ -825,4 +825,289 @@ public function __construct(array $field1)
+        $dependsOnTheseNamespaces->evaluate($cd[0], $violations, 'we want to add this rule for our software');
+
+        $this->assertCount(1, $violations);
+    }
+
+    public function test_it_handles_typed_arrays_in_properties_with_list_syntax(): void
+    {
+        $code = <<< 'EOF'
+<?php
+namespace Domain\Foo;
+
+use Application\MyDto;
+
+class MyClass
+{
+    /**
+     * @var list<MyDto>
+     */
+    private array $dtoList;
+}
+EOF;
+
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $fp = FileParserFactory::createFileParser(TargetPhpVersion::create('7.1'));
+        $fp->parse($code, 'relativePathName');
+        $cd = $fp->getClassDescriptions();
+
+        $violations = new Violations();
+
+        $dependsOnTheseNamespaces = new DependsOnlyOnTheseNamespaces('Domain');
+        $dependsOnTheseNamespaces->evaluate($cd[0], $violations, 'we want to add this rule for our software');
+
+        $this->assertCount(1, $violations);
+    }
+
+    public function test_it_handles_typed_arrays_in_properties_with_legacy_syntax(): void
+    {
+        $code = <<< 'EOF'
+<?php
+namespace Domain\Foo;
+
+use Application\MyDto;
+
+class MyClass
+{
+    /**
+     * @var MyDto[]
+     */
+    private array $dtoList;
+}
+EOF;
+
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $fp = FileParserFactory::createFileParser(TargetPhpVersion::create('7.1'));
+        $fp->parse($code, 'relativePathName');
+        $cd = $fp->getClassDescriptions();
+
+        $violations = new Violations();
+
+        $dependsOnTheseNamespaces = new DependsOnlyOnTheseNamespaces('Domain');
+        $dependsOnTheseNamespaces->evaluate($cd[0], $violations, 'we want to add this rule for our software');
+
+        $this->assertCount(1, $violations);
+    }
+
+    public function test_it_handles_typed_arrays_in_method_params_with_generics_syntax(): void
+    {
+        $code = <<< 'EOF'
+<?php
+namespace Domain\Foo;
+
+use Application\MyDto;
+
+class MyClass
+{
+    /**
+     * @param array<int, MyDto> $dtoList
+     */
+    public function __construct(array $dtoList)
+    {
+    }
+}
+EOF;
+
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $fp = FileParserFactory::createFileParser(TargetPhpVersion::create('7.1'));
+        $fp->parse($code, 'relativePathName');
+        $cd = $fp->getClassDescriptions();
+
+        $violations = new Violations();
+
+        $dependsOnTheseNamespaces = new DependsOnlyOnTheseNamespaces('Domain');
+        $dependsOnTheseNamespaces->evaluate($cd[0], $violations, 'we want to add this rule for our software');
+
+        $this->assertCount(1, $violations);
+    }
+
+    public function test_it_handles_typed_arrays_in_method_params_with_list_syntax(): void
+    {
+        $code = <<< 'EOF'
+<?php
+namespace Domain\Foo;
+
+use Application\MyDto;
+
+class MyClass
+{
+    /**
+     * @param list<MyDto> $dtoList
+     */
+    public function __construct(array $dtoList)
```

Comment:
not sure it is worth it, but what if we add a test case with multiple param?

Reply:
@micheleorselli yes, it can helps. I just added a test with a method with multiple params: 2 primitive types (one with native type and one with PHPDoc type) and 2 array types (one with a violation and one with a class allowed in the rules).